src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/HalfConverter.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    /// Provides a type converter to convert half-precision, floating point number objects to and
+    /// from various other representations.
+    /// </summary>
+    public class HalfConverter : BaseNumberConverter
+    {
+        /// <summary>
+        /// Determines whether this editor will attempt to convert hex (0x or #) strings
+        /// </summary>
+        internal override bool AllowHex => false;
+
+        /// <summary>
+        /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
+        /// </summary>
+        internal override Type TargetType => typeof(Half);
+
+        /// <summary>
+        /// Convert the given value to a string using the given radix
+        /// </summary>
+        internal override object FromString(string value, int radix)
+        {
+            Debug.Assert(radix == 10);
+            Debug.Assert(value is not null);
+            return Half.Parse(value!, CultureInfo.CurrentCulture);
```

Comment:
This method is only ever called with `radix == 16`, and only if `AllowHex == true`:

https://github.com/dotnet/runtime/blob/b098b6ffb9274fd9f9fd6db943e291ce9909af84/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/BaseNumberConverter.cs#L61-L68

1. This is dead code.
2. The assert doesn't match the only places this method could be called from

I think it would be better to just `throw new NotImplementedException()`. We could change Single and Double converters as well, since they don't support `AllowHex` either.

Alternatively, we could change the method from `abstract` to `virtual` in the base, and have the base throw.

Reply:
> I think it would be better to just throw new NotImplementedException(). We could change Single and Double converters as well, since they don't support AllowHex either.

Notably this support is approved just not implemented yet. It's on the backlog of support to add.

But 👍 to not having dead code today. We can track it as something to update when https://github.com/dotnet/runtime/issues/1630 is implemented

Reply:
I'll throw NotImplementedException for the Half type as `FromString` is not expected to be called there. But I'll not touch `Double` or `Single` converter code. I don't think it will make any difference changing it.